backend/app/src/game-socket/GameSocketEvents.ts

```diff
@@ -0,0 +1,44 @@
+
+import
+{
+	OnGatewayConnection,
+	OnGatewayDisconnect,
+	OnGatewayInit,
+	WebSocketGateway,
+	WebSocketServer
+} from "@nestjs/websockets";
+import { Server, Socket } from "socket.io";
+
+
+@WebSocketGateway(
+{
+	cors:
+	{
+		origin: "*"
+	},
+	// namespace: "/game-engine-v2d"
+})
+export class GameSocketEvents
+	implements OnGatewayInit, OnGatewayConnection, OnGatewayDisconnect
```

Comment:
We forgot to implement the class lifecycle decorator that handle respectively afterInit, handleConnection, and handleDisconnect